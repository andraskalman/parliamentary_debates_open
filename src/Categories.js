import React from 'react';
import { Checkbox } from 'react-bootstrap';

const Categories = ({ categories, onChange }) => (
  <ul>
    {Object.keys(categories).map(category => (
      <li key={category}>
        <Checkbox onChange={event => onChange({
          category,
          value: event.target.checked}
        )} checked={categories[category]}>
          {category}
        </Checkbox>
      </li>
    ))}
  </ul>
)

export default Categories
