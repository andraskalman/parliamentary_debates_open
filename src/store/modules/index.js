import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import search from './search'
import help from './help'

export default combineReducers({
  routing: routerReducer,
  search,
  help
})
