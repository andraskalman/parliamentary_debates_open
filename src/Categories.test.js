import React from 'react';
import ReactDOM from 'react-dom';
import Categories from './Categories';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store, { history } from './store'
import { ConnectedRouter } from 'react-router-redux'


it('renders correctly', () => {
  const tree = renderer
    .create(<Categories categories={{"foo": true, "bar": false}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
