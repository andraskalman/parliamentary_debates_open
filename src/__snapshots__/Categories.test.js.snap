// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <li>
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          checked={true}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        foo
      </label>
    </div>
  </li>
  <li>
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          checked={false}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        bar
      </label>
    </div>
  </li>
</ul>
`;
